### DISCUSSION TITLE
Who decides which agent gets executed next?

### DISCUSSION THREAD

**Discussion by Yanni8** [microsoft/autogen#1466] [Q&A] [answered]
I know that there is the possibility of choosing between different ways to choose the next agent:

https://microsoft.github.io/autogen/docs/reference/agentchat/groupchat#groupchat-objects

What I don't understand is, how auto works. What Agent (LLM) decides who's next, and based on what information?

**Comment by ekzhu** [comment, marked as answer]
The group chat manager uses this method to decide when set to `auto`: https://github.com/microsoft/autogen/blob/main/autogen/agentchat/groupchat.py#L92

**Reply by Yanni8** [reply]
Thanks exactly what I searched for :)
